view/src/components/DiffViewWrapper.jsx

```diff
@@ -0,0 +1,236 @@
+} from 'react';
+import { PropTypes } from 'prop-types';
+import {
+  Diff,
+  Hunk,
+  parseDiff,
+  expandFromRawCode,
+  Decoration,
+  textLinesToHunk,
+  tokenize,
+  findChangeByOldLineNumber,
+  findChangeByNewLineNumber,
+  getChangeKey,
+  getCorrespondingOldLineNumber,
+} from 'react-diff-view';
+import refractor from 'refractor';
+
+/** Show whole function if function has max this amount of line. */
+const MAX_LINES_THRESHOLD = 25;
+/** Amount of lines to show around calling function
+ * if function is too long to show whole. */
+const LINES_OF_CONTEXT = 10;
+
+function isFunctionLong(startLine, endLine) {
+  return endLine - startLine >= MAX_LINES_THRESHOLD;
+}
+
+/**
+ * Creates a file structure necessary for visualisation of two functions
+ * with empty diff. Creates one hunk containing just a single line:
+ * the line numbered oldStart in oldCode. newStart is the number of
+ * the corresponding line in the new file.
+ * @param {String} oldCode - Content of file containing old version of function
+ * @returns Returns file structure containing created hunk.
+ */
+function createFileStructure(oldCode, oldStart, newStart) {
+  const linesOfCode = oldCode.split('\n');
+  const lineOfCode = linesOfCode[oldStart - 1];
+  const hunkFromLine = textLinesToHunk([lineOfCode], oldStart, newStart);
+  const file = {
+    type: 'modify',
+    hunks: [hunkFromLine],
+  };
+  return file;
+}
+
+/**
+ * Transform lines of code (to be highlighted) to array which Diff accepts.
+ * @param {Number} linesToShow - Lines (old, new) of code which should be highlighted.
+ * @returns Returns array of lines to be highlighted acceptable by Diff.
+ */
+function getLinesToHighlight(linesToShow, hunks) {
+  if (linesToShow) {
+    const oldChange = findChangeByOldLineNumber(hunks, linesToShow[0]);
+    const newChange = findChangeByNewLineNumber(hunks, linesToShow[1]);
+    if (oldChange && newChange) {
+      const oldChangeKey = getChangeKey(oldChange);
+      const newChangeKey = getChangeKey(newChange);
+      if (oldChangeKey === newChangeKey) return [oldChangeKey];
+      return [oldChangeKey, newChangeKey];
+    }
+  }
+  return [];
+}
+
+function getSyntaxHighlightedTokens(hunks) {
+  const options = {
+    highlight: true,
+    refractor,
+    language: 'c',
+  };
+  return tokenize(hunks, options);
+}
+
+/**
+ * A component for visualisation of source code of two versions of function.
+ * For the function are shown lines of code:
+ * - which contains differences,
+ * - first and last line of the function.
+ * If the function is calling another function (linesToShow is set) than
+ * - if the function is short (<= MAX_LINES_THRESHOLD) than code of the whole function is shown,
+ * - else the line with the call is shown
+ *   and certain amount (LINES_OF_CONTEXT) of lines above and below the call.
+ * For the lines which are not shown are rendered buttons that allow user to show these lines
+ * by clicking on the button.
```

Comment:
Looks good, just minor rewordings:
```suggestion
 * A component for visualisation of source code of two versions of a function.
 * For the function, we show:
 * - lines containing differences,
 * - the first and the last line of the function.
 * If the function is calling another function (`linesToShow` is set), then
 * - if the function is short (<= MAX_LINES_THRESHOLD), the entire function is shown,
 * - otherwise, only the line with the call surrounded by a certain amount (LINES_OF_CONTEXT) of lines is shown.
 * For the lines which are not shown, we render buttons that allow user to show these lines.
```

Reply:
Thanks, I added blank lines for separation of enumerations.